docs: clarify aside complementary role mapping (#44629)
mdn/content: files/en-us/web/accessibility/aria/reference/roles/complementary_role/index.md

The `complementary` role is [a landmark](/en-US/docs/Web/Accessibility/ARIA/Guides/Techniques#landmark_roles) role. Landmarks can be used by assistive technology to quickly identify and navigate to large sections of the document. Content listed within a container with the `complementary` landmark role should make sense if separated from the main content of the document.

> [!NOTE]

> > Using the {{HTMLElement('aside')}} element will usually communicate a section has a role of `complementary`. When an `<aside>` is nested in sectioning content, it maps to `complementary` only if it has an accessible name. Developers should always prefer using the correct semantic HTML element over using ARIA.

Reviewer: We can make it shorter. does this work?

```suggestion
> Using the {{HTMLElement('aside')}} element will communicate a section has a role of `complementary` when the `<aside>` has an accessible name and is nested in sectioning content. Developers should always prefer using the correct semantic HTML element over using ARIA.
```

### Prefer HTML

> Using the {{HTMLElement('aside')}} element will usually communicate that the element has a role of `complementary`. If an `<aside>` is nested in sectioning content, it maps to `complementary` only if it has an accessible name. If possible, prefer using the semantic `<aside>` element instead of the `complementary` role.

Reviewer: ```suggestion
Using the {{HTMLElement('aside')}} element will communicate a section has a role of `complementary` when the `<aside>` has an accessible name and is nested in sectioning content. If possible, prefer using the semantic `<aside>` element instead of the `complementary` role.
```
